### PR TITLE
Add constant folding for `lq.quantize` and `lq.dequantize`

### PR DESCRIPTION
**Pull request by lgeiger**
## What do these changes do?
This PR adds constant folders for `lq.quantize` and `lq.dequantize` ops.

This PR also includes a few minor changes from our internal fork to bring them inline with each other.

## How Has This Been Tested?
Added MLIR Filecheck

Co-Authored-By: Tom Bannink <Tombana@users.noreply.github.com>

### CODE REVIEW

**Review by Tombana — APPROVED** [review]
Looks good to me, but maybe somebody else should also review it.

image

File: larq_compute_engine/core/types.h [hunk 1]
```diff
@@ -17,6 +19,12 @@ namespace core {
 #define LCE_UNLIKELY(condition) (condition)
 #endif
 
+#if defined(__GNUC__)
+#define FORCE_INLINE __attribute__((always_inline)) inline
+#else
+#define FORCE_INLINE inline
+#endif
```
Comment (CNugteren) [inline larq_compute_engine/core/types.h:26]: Maybe it is my search failing, but I don't think you use `FORCE_INLINE` somewhere?

**Review by CNugteren — APPROVED** [review]
Just a small comment, otherwise I approve. The name of the PR was a bit confusing to me though, I was going through the code to look for the place where you added a folder or moved something from one folder to another, but then it turns out it was about folding instead of folders (as in paths) :confused: :wink:

**Comment by lgeiger** [comment]
> The name of the PR was a bit confusing to me though, I was going through the code to look for the place where you added a folder or moved something from one folder to another, but then it turns out it was about folding instead of folders (as in paths)

Good point, changed the title.

File: larq_compute_engine/core/types.h (hunk 1 above)
Comment (lgeiger) [reply larq_compute_engine/core/types.h:26]: Indeed it is not used anywhere at the moment. I mainly added it to sync `larq_compute_engine/core/types.h` with our internal fork to reduce confusing for us in the future. But I am happy to remove it, if you prefer.

Comment (Tombana) [reply larq_compute_engine/core/types.h:26]: Yea i'd opt to keep public and private fork as much in sync as possible.

**Pull request merged**